node/src/database/db_migrations.rs

```diff
@@ -169,124 +184,96 @@ impl DbMigratorReal {
+        mig_utils: &dyn DBMigrationUtilities,
     ) -> Vec<&'a (dyn DatabaseMigration + 'a)> {
-        let updates_to_process = list_of_updates
+        mig_utils.panic_if_the_version_found_is_too_high(mismatched_schema);
+        list_of_updates
             .iter()
-            .skip_while(|entry| entry.old_version().ne(&mismatched_schema))
+            .skip_while(|entry| entry.old_version() != mismatched_schema)
             .map(Self::deref)
-            .collect::<Vec<&(dyn DatabaseMigration + 'a)>>();
-        let _ = Self::check_out_quantity_of_those_remaining(
-            mismatched_schema,
-            updates_to_process.len(),
-        );
-        updates_to_process
+            .collect::<Vec<&(dyn DatabaseMigration + 'a)>>()
     }
 
-    fn deref<'a>(value: &'a &dyn DatabaseMigration) -> &'a dyn DatabaseMigration {
+    fn deref<'a, T: ?Sized>(value: &'a &T) -> &'a T {
         *value
     }
 
-    fn process_items_with_dirty_references<'a>(
-        first: Option<&'a &dyn DatabaseMigration>,
-        second: Option<&&&dyn DatabaseMigration>,
-    ) -> (&'a dyn DatabaseMigration, String) {
-        let first = *first.expect_v("migration record");
-        let identity_of_the_second = Self::identify_the_next_record(second, first);
-        (first, identity_of_the_second)
-    }
-
-    fn identify_the_next_record(
-        subject: Option<&&&dyn DatabaseMigration>,
-        current_record: &dyn DatabaseMigration,
-    ) -> String {
-        if let Some(next_higher) = subject {
-            next_higher.old_version().to_string()
-        } else {
-            (current_record.old_version() + 1).to_string()
-        }
-    }
-
-    fn check_out_quantity_of_those_remaining(mismatched_schema: usize, count: usize) {
-        if count == 0 {
-            panic!("Database claims to be more advanced ({}) than the version {} which is the latest released.", mismatched_schema, CURRENT_SCHEMA_VERSION)
-        }
-    }
-
-    fn dispatch_bad_news(&self, versions: &str, error: rusqlite::Error) -> Result<(), String> {
-        let error_message = format!("Updating database {} failed: {:?}", versions, error);
-        warning!(self.logger, "{}", &error_message);
+    fn dispatch_bad_news(
+        &self,
+        current_version: usize,
+        error: rusqlite::Error,
+    ) -> Result<(), String> {
+        let error_message = format!(
+            "Updating database from version {} to {} failed: {:?}",
+            current_version,
+            current_version + 1,
+            error
+        );
+        error!(self.logger, "{}", &error_message);
         Err(error_message)
     }
 
-    fn context_between_two_versions(first: usize, second: &str) -> String {
-        format!("from version {} to {}", first, second)
-    }
-
-    fn log_success(&self, versions: &str) {
-        info!(self.logger, "Database successfully updated {}", versions)
+    fn log_success(&self, current_version: &usize) {
```

Comment:
Maybe this variable name could be changed...if the database was successfully updated, then the current version is `current_version + 1`, rather than `current_version.`

Also, `&usize` is no smaller than `usize`.